### PR TITLE
feat(paddle-backend): support list in `quantile` function's `q` parameter

### LINKED ISSUE #23608: [Bug]:  Support for List in `quantile` function's `q` parameter for Ivy Paddle Backend
### Bug Explanation

In the native Paddle framework, the [quantile](https://www.paddlepaddle.org.cn/documentation/docs/en/api/paddle/quantile_en.html#quantile) function accepts an int, float, or list for its` q` parameter. However, in the Ivy paddle backend, only  `float` are currently accepted.

### Steps to Reproduce Bug

```py
import ivy 
import paddle

ivy.set_backend("paddle")
y = paddle.arange(0, 8 ,dtype="float32").reshape([4, 2])
x = ivy.quantile(y, [0.3, 0.5], axis=0) # throws error
print(x)
# paddle native 
z = paddle.quantile(y, q=[0.3, 0.5], axis=0)
print(z) # works

```

### Environment

Linux, Docker, vscode

### Ivy Version

1.1.9

### Backend

- [ ] NumPy
- [ ] TensorFlow
- [ ] PyTorch
- [ ] JAX

### Device

CPU

### PR DESCRIPTION
**Pull request by akshatvishu**
# PR Description

Added the ability for the `quantile` function in the Ivy paddle backend to accept a `list` as a type for its `q` parameter. This enhancement ensures consistency with the native Paddle framework.

## Related Issue

Close #23608

## Additional Information

All test for `ivy.quantile` fails locally(even if you run it without making these changes ). So, it might need some refactoring!
 
```bash
python3 -m pytest -v ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py::test_quantile
```
image

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [ ] Did you follow the steps we provided?

### Socials:

[fun](https://newskit.social/blog/posts/glg)

### CODE REVIEW

**Comment by MahmoudAshraf97** [comment]
Hello, I guess we should start by fixing the existing function first before extending its functionality

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.
